### ISSUE TITLE
linespacing of multiline texts.

### ISSUE THREAD

**Issue by leejjoon** [matplotlib/matplotlib#2078]
The linespacing of multiline text in the current master branch needs some improvement.

Consider a following example, 

``` python
import matplotlib.pyplot as plt
plt.figure(figsize=(7,3))
plt.text(0.2, 0.5, "TpTpTp\n$M$\nTpTpTp", size=20,
         ha="center", va="top")

plt.text(0.5, 0.5, "TpTpTp\n$M^{M^{M^{M}}}$\nTpTpTp", size=20,
         ha="center", va="top")

plt.text(0.8, 0.5, "TpTpTp\n$M_{q_{q_{q}}}$\nTpTpTp", size=20,
         ha="center", va="top")

plt.xlim(0, 1)
plt.ylim(0, 0.8)
plt.show()
```

Current master (and mdboom's issue2070 branch) produces following.

![test_multiline_linespacing_master](https://f.cloud.github.com/assets/95962/576940/449c160a-c81b-11e2-8442-7cd3392fb78c.png)

Note that superscrips and subscripts overwarp with the line above and below.

In mpl 1.2,

![test_multiline_linespacing_mpl1 2](https://f.cloud.github.com/assets/95962/576943/7633e0ee-c81b-11e2-9cfb-9a67b9a546a5.png)

The second looks good, though 3rd still overwrapps.

The desired behavior is (at least to me)

![test_multiline_linespacing_okay](https://f.cloud.github.com/assets/95962/576944/857b262a-c81b-11e2-8a2c-bc49cc3a67e4.png)

I have a working solution based on the v1.2.x branch. Once mdboom's issue2070 branch is merged into master, I will rebase and make a PR.

**Comment by petehuang** [comment]
In 1.5.3 I got something like the following.

![2078](https://cloud.githubusercontent.com/assets/1134834/21547629/32c981a8-cd9c-11e6-95b0-9ec5434f4ed9.png)

I think we can close this.

**Comment by tacaswell** [comment]
Indeed, nested super/subscripts even work better now!
